File: create_wp_post.py
import requests

# Impostare le informazioni dell'endpoint della REST API di WordPress
url = 'https://example.com/wp-json/wp/v2/posts'
headers = {'Content-Type': 'application/json'}
auth = ('username', 'password') # sostituire con le credenziali dell'utente autorizzato

# Impostare i dati del post da creare
data = {
    'title': 'Titolo del post',
    'content': 'Testo del post',
    'status': 'publish', # Stato del post, ad esempio: "publish", "draft", "pending", "private"
    'categories': [1, 2], # ID delle categorie del post, sostituire con gli ID delle categorie desiderate
    'tags': ['tag1', 'tag2'], # Etichette del post, sostituire con le etichette desiderate
    'meta': {
        'key1': 'valore1',
        'key2': 'valore2'
    } # Altri metadati del post, sostituire con i propri metadati
}

# Invio della richiesta POST alla REST API di WordPress
response = requests.post(url, headers=headers, auth=auth, json=data)

# Stampa della risposta della richiesta
print(response.json())

File: create_wp_post.php
<?php

// Imposta l'URL del sito WordPress e l'endpoint REST API
$site_url = 'http://tuosito.com';
$rest_api_endpoint = '/wp-json/wp/v2/posts';

// Imposta il titolo e il contenuto del post
$post_title = 'Titolo del post';
$post_content = 'Contenuto del post';

// Imposta i dati del post da inviare tramite la REST API
$data = array(
    'title' => $post_title,
    'content' => $post_content,
    'status' => 'publish'
);

// Converti i dati in formato JSON
$json_data = json_encode($data);

// Imposta le opzioni della richiesta HTTP
$options = array(
    'http' => array(
        'method' => 'POST',
        'header' => "Content-type: application/json\r\n",
        'content' => $json_data
    )
);

// Crea il contesto della richiesta HTTP
$context = stream_context_create($options);

// Invia la richiesta HTTP alla REST API di WordPress per creare il post
$response = file_get_contents($site_url . $rest_api_endpoint, false, $context);

// Stampa la risposta della REST API di WordPress
echo $response;
